src/AggregateFunctions/UniquesHashSet.h

```diff
@@ -108,7 +108,13 @@ class UniquesHashSet : private HashTableAllocatorWithStackMemory<(1ULL << UNIQUE
     inline size_t buf_size() const           { return 1ULL << size_degree; } /// NOLINT
     inline size_t max_fill() const           { return 1ULL << (size_degree - 1); } /// NOLINT
     inline size_t mask() const               { return buf_size() - 1; }
-    inline size_t place(HashValue x) const { return (x >> UNIQUES_HASH_BITS_FOR_SKIP) & mask(); }
+
+    inline size_t place(HashValue x) const {
+        if constexpr (std::endian::native == std::endian::little)
+            return (x >> UNIQUES_HASH_BITS_FOR_SKIP) & mask();
+        else
+            return (std::byteswap(x) >> UNIQUES_HASH_BITS_FOR_SKIP) & mask();
```

Comment:
Hash value is the same as it has already been fixed by https://github.com/ClickHouse/ClickHouse/pull/48764. 
This PR mainly raised to solve the issue when query result needs to be merged together. For example, in fvt 02429_combinators_in_array_reduce, select arrayReduce('uniqStateForEach', [[1], [2]]);  would produce  ['\0e,4']  ,  instead of  ['\0,4e']. 
Please suggest if there is a better place to solve this problem.